srcjs/input_binding_text.js

```diff
@@ -27,8 +27,27 @@ $.extend(textInputBinding, {
     if (data.hasOwnProperty('value'))
       this.setValue(el, data.value);
 
-    if (data.hasOwnProperty('label'))
-      $(el).parent().find('label[for="' + $escape(el.id) + '"]').text(data.label);
+
+    var labelTag = $(el).parent().find('label[for="' + $escape(el.id) + '"]');
+    var hasLabelTag = labelTag.length > 0;
+
+    // If data.label exists, then we may need to insert a label
+    // tag into the DOM. If it doesn't exist, then the label tag
+    // should be removed to be more consistent with the behavior
+    // of `textInput(label = NULL)`
+    if (data.hasOwnProperty('label')) {
+
+      if (hasLabelTag) {
+        labelTag.text(data.label);
+      } else {
+        $('<label for="' + $escape(el.id) + '">' + data.label + '</label>').insertBefore(el);
```

Comment:
I think this `data.label` might need to be escaped. For example, what if its contents are `abc<br>xyz`?